content/root/best-practices-guide.textile

```diff
@@ -0,0 +1,97 @@
+
+h4. 4. Handling connection and channel state&#45;change events
+
+* Ably's Realtime SDKs automatically handle disruptions such as network disconnections, so you don't need to do anything. The library emits "connection":/realtime/connection#connection-states and "channel":/realtime/channels-messages#channel-stateschannel state change events that you can listen to in case you'd like be informed about what it's doing (for example, if you want to know whenever the library has lost message continuity on a channel due to being disconnected for more than two minutes). But there's no need to take any particular action in response to any of these, the library will automatically reconnect and reattach to channels after the network disruption is over.
+
+h4. 5. Using API key on the client side
+
+* Ably's "Basic Authentication":/general/authentication#basic-authentication method is the easiest way to authenticate your client on the Ably Realtime platform. However, it directly exposes the API key in the source code. When you use it on the client side, any entity that interacts with the client could view your API key and potentially misuse it. If you can run an auth server, we recommend using "Token Authentication":/general/authentication#token-authentication on the client side. "Read more":/general/authentication#selecting-auth about selecting an authentication mechanism.
+
+h4. 6. Ensuring tokens are auto&#45;renewable
+
+* Tokens are short lived and need to be constantly renewed. Rather than getting a token from your auth servers yourself and initializing the Client Library SDK with it, we recommend you instead initialize the SDK with an authUrl or authCallback. It will use this information to automatically request a token from your auth server's endpoint whenever it needs a new one, thus seamlessly switching to the new token without disconnecting, hence ensuring an uninterrupted service.
+
+h4. 7. Choosing between Realtime/ REST libraries and Protocol Adapters 
+
+* Want to maintain a persistent connection with Ably's Realtime platform, so you can subscribe to messages sent on a channel, or be notified whenever the presence set changes? Use the "realtime library":/realtime.
+* Want to just publish data to the platform, don't care about subscribing, and don't want to manage channel attachment? Use the "REST library":/rest.
+* Want to publish at very high volumes or with the lowest possible latency? "The realtime libary can publish at higher volumes (due to message pipelining) and lower latencies than is possible with the REST library":https://support.ably.io/solution/articles/3000060319-should-i-use-the-rest-or-realtime-library.
```

Comment:
Typos.  I am not sure we've really quantified high volumes or low latencies.  I think in most cases REST is still the preferred route, but in some cases a realtime publish is preferred.